test/parallel/test-tls-keylog-tlsv13.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const events = require('events');
+const tls = require('tls');
+const fixtures = require('../common/fixtures');
+
+async function main() {
+  // Create server / client pair
+  const server = tls.createServer({
+    key: fixtures.readSync('/keys/agent2-key.pem'),
+    cert: fixtures.readSync('/keys/agent2-cert.pem'),
+    minVersion: 'TLSv1.3',
+    maxVersion: 'TLSv1.3',
+  });
+  server.listen();
+  await events.once(server, 'listening');
+  server.once('connection', () => server.close());
+
+  const client = tls.connect({
+    port: server.address().port,
+    rejectUnauthorized: false,
+  });
+
+  // Record keylog lines
+  const lines = { client: new Set(), server: new Set() };
+  client.on('keylog', (line) => lines.client.add(line));
+  server.on('keylog', (line) => lines.server.add(line));
+
+  // Wait for handshake, test that same lines were produced
+  await Promise.all([ events.once(server, 'secureConnection'),
+                      events.once(client, 'secureConnect') ]);
+  assert.deepStrictEqual(lines.client, lines.server);
+  assert.strictEqual(lines.client.size, 5);
+
+  client.destroy();
+}
+
+main().catch((error) => process.nextTick(() => {
```

Comment:
By the way, we sometimes use `main().then(common.mustCall())` – it checks that no error has occurred, but also that the async function has completed by the time that the process exits.

Reply:
I was looking for something like `common.mustCall()`, thanks!